smaug/python/graph.py

```diff
@@ -110,23 +110,21 @@ def create_unique_name(self, name, mark_as_used=True):
     """ Create a unique name for the node.
 
     Args:
-      name: The node's name.
+      name: The base name used to create the unique name.
+      mark_as_used: Mark the unique name as used so if someone wants to call
+        create_unique_name(unique_name), a different name will be created.
     """
-    if name not in self._node_names:
-      self._node_names[name] = 0
-      return name
-    else:
+    new_name = name
+    if name in self._node_names:
       while True:
         self._node_names[name] += 1
         new_name = "%s_%d" % (name, self._node_names[name])
         # Make sure the new name is not already used.
         if new_name not in self._node_names:
           break
-      # Mark the new name as used in case someone wants to call
-      # create_unique_name("name_1").
-      if mark_as_used:
-        self._node_names[new_name] = 0
-      return new_name
+    if mark_as_used:
```

Comment:
So the bug was that mark_as_used would only be checked if the name was already in the set of used names? If so, can you update the PR description?

Reply:
Yes. Done.